src/main.js

```diff
@@ -20,13 +22,43 @@ let load = base.loader({
   },
 
   diagnostics: {
-    requires: ['monitor'],
-    setup: async ({monitor}) => { 
+    requires: ['cfg','monitor'],
+    setup: async ({cfg, monitor}) => { 
       // Running tests
-      let result = await TestSpawn.runTests(monitor);
+      let data = await TestSpawn.runTests();
+      /*
+       * Reporting via notify.
+       */
+      debug(data);
+      let result = data.result;
+      let keys = data.keys;
+      //TODO: Add notify email
+      try{
+        let notify  = new taskcluster.Notify({
+            credentials: cfg.taskcluster.credentials
+        });
+        let subject = 'Diagnostics results: ' + (new Date()).toJSON();
+        let content = [`## Number of passing tests:  ${result.pass.length}`];
+        result.pass.forEach(pass => content.push("* " + pass));
+        content.push(`## Number of failing tests: ${result.fail.length}`);
+        result.fail.forEach(fail => content.push("* " + fail));
+        content = content.join("\n");
+        let link = {
+          text: "View raw logs",
+          href: "https://taskcluster-diagnostic-logs.s3.amazonaws.com/" + keys.raw_key
+        }
+        cfg.emails.forEach(async address => {
+          debug('sending email to ' + address);
+          await notify.email({ address, subject, content, link });
+        });
+      }catch(e){
```

Comment:
Do we really want all of this inside a big try-catch with a `console.log(e)` at the end? This seems like not a great thing to do unless there's a good reason. If there's a good reason, add it in a comment and if not, just remove the try-catch.